cuML/src/solver/solver.cu

```diff
@@ -340,5 +338,158 @@ void sgdPredictBinaryClass(const double *input, int n_rows, int n_cols,
 
 }
 
+void cdFit(float *input,
+		   int n_rows,
+		   int n_cols,
+		   float *labels,
+		   float *coef,
+		   float *intercept,
+		   bool fit_intercept,
+		   bool normalize,
+		   int epochs,
+		   int loss,
+		   float alpha,
+		   float l1_ratio,
+		   bool shuffle,
+		   float tol) {
+
+	ASSERT(loss == 0,
+			"Parameter loss: Only SQRT_LOSS function is supported for now");
+
+	ML::loss_funct loss_funct = ML::loss_funct::SQRD_LOSS;
+
+	cublasHandle_t cublas_handle;
+	CUBLAS_CHECK(cublasCreate(&cublas_handle));
+
+	cusolverDnHandle_t cusolver_handle = NULL;
+	CUSOLVER_CHECK(cusolverDnCreate(&cusolver_handle));
+
+	cudaStream_t stream;
+	CUDA_CHECK(cudaStreamCreate(&stream));
+
+	cdFit(input,
+			   n_rows,
+			   n_cols,
+			   labels,
+			   coef,
+			   intercept,
+			   fit_intercept,
+			   normalize,
+			   epochs,
+			   loss_funct,
+			   alpha,
+			   l1_ratio,
+			   shuffle,
+			   tol,
+			   stream,
+			   cublas_handle,
+			   cusolver_handle);
+
+	CUBLAS_CHECK(cublasDestroy(cublas_handle));
+	CUSOLVER_CHECK(cusolverDnDestroy(cusolver_handle));
+	CUDA_CHECK(cudaStreamDestroy(stream));
+
+}
+
+void cdFit(double *input,
+		   int n_rows,
+		   int n_cols,
+		   double *labels,
+		   double *coef,
+		   double *intercept,
+		   bool fit_intercept,
+		   bool normalize,
+		   int epochs,
+		   int loss,
+		   double alpha,
+		   double l1_ratio,
+		   bool shuffle,
+		   double tol) {
+
+	ASSERT(loss == 0,
+			"Parameter loss: Only SQRT_LOSS function is supported for now");
+
+	ML::loss_funct loss_funct = ML::loss_funct::SQRD_LOSS;
+
+	cublasHandle_t cublas_handle;
+	CUBLAS_CHECK(cublasCreate(&cublas_handle));
+
+	cusolverDnHandle_t cusolver_handle = NULL;
+	CUSOLVER_CHECK(cusolverDnCreate(&cusolver_handle));
+
+	cudaStream_t stream;
+	CUDA_CHECK(cudaStreamCreate(&stream));
+
+	cdFit(input,
+			   n_rows,
+			   n_cols,
+			   labels,
+			   coef,
+			   intercept,
+			   fit_intercept,
+			   normalize,
+			   epochs,
+			   loss_funct,
+			   alpha,
+			   l1_ratio,
+			   shuffle,
+			   tol,
+			   stream,
+			   cublas_handle,
+			   cusolver_handle);
+
+	CUBLAS_CHECK(cublasDestroy(cublas_handle));
+	CUSOLVER_CHECK(cusolverDnDestroy(cusolver_handle));
+	CUDA_CHECK(cudaStreamDestroy(stream));
+
+}
+
+void cdPredict(const float *input, int n_rows, int n_cols, const float *coef,
+		float intercept, float *preds, int loss) {
+
+	ML::loss_funct loss_funct = ML::loss_funct::SQRD_LOSS;
+	if (loss == 0) {
+		loss_funct = ML::loss_funct::SQRD_LOSS;
+	} else {
+		ASSERT(false,
+			"glm.cu: other functions are not supported yet.");
```

Comment:
Not a big thing, but might want to add the function to the string so that it is perfectly clear where things are failing.

Reply:
I think we get the line number when we get the error, no?